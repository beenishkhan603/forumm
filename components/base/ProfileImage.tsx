import { useState } from 'react'
import User from '@public/images/User.svg'
import Image from 'next/image'

const ProfileImage = ({
  imageUrl,
  className = '',
  onClick,
  size = 'md',
  activityStatus,
  tooltip,
}: {
  imageUrl?: string | null | undefined
  size?: 'xs' | 'sm' | 'md' | 'lg'
  className?: string
  onClick?: (e: any) => void
  activityStatus?: boolean | undefined | null
  tooltip?: string[]
}) => {
  const [profileImageError, setProfileImageError] = useState(false)
  const [showTooltip, setShowTooltip] = useState(false)

  const sizes = {
    xs: 'w-6 h-6',
    sm: 'w-8 h-8',
    md: 'w-10 h-10',
    lg: 'w-16 h-16',
  }

  const handleMouseEnter = () => setShowTooltip(true)
  const handleMouseLeave = () => setShowTooltip(false)

  return (
    <div
      className={`relative flex-shrink-0 ${sizes[size]} ${className} ${
        onClick ? 'cursor-pointer' : ''
      }`}
      onMouseEnter={handleMouseEnter}
      onMouseLeave={handleMouseLeave}
    >
      {profileImageError || !imageUrl ? (
        <User
          className={`${sizes[size]} rounded-full  bg-white`}
          onClick={onClick}
        />
      ) : (
        <Image
          src={imageUrl}
          width={32}
          height={32}
          alt="Profile Image"
          className={`${sizes[size]} bg-contain bg-center rounded-full object-cover `}
          onError={() => setProfileImageError(true)}
          onClick={onClick}
        />
      )}
      {/* {activityStatus != null && ( */}
      {/*   <div */}
      {/*     className={`w-2.5 h-2.5 rounded-full -mt-2 -mr-0.5 right-0 absolute ${ */}
      {/*       activityStatus */}
      {/*         ? 'bg-green-600 opacity-90' */}
      {/*         : 'bg-gray-500 opacity-70' */}
      {/*     }`} */}
      {/*   ></div> */}
      {/* )} */}
      {tooltip && showTooltip && (
        <div
          style={{ width: '150px' }}
          className="absolute pt-4 text-center bg-gray-700 text-white rounded-md shadow-lg -top-14 left-1/2 transform -translate-x-1/2 -translate-y-1/2 tooltip-arrow"
        >
          <p className="text-md mb-1 font-bold ml-4 mr-4">{tooltip[0]}</p>
          {tooltip.slice(1).map((row) => (
            <p key={row} className="text-xs mb-1 ml-4 mr-4">
              {row}
            </p>
          ))}
          <p></p>
        </div>
      )}
    </div>
  )
}

export default ProfileImage
