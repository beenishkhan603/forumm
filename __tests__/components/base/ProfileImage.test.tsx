import ProfileImage from '@components/base/ProfileImage'
import { render } from '@testing-library/react'

it.each([true, false, undefined])(
  'renders profile image unchanged',
  (status) => {
    const { container } = render(
      <ProfileImage
        size="md"
        imageUrl={'/foo.png'}
        className="foo bar"
        activityStatus={status}
      />
    )
    expect(container).toMatchSnapshot()
  }
)
