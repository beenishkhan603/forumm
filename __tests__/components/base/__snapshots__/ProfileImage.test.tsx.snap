// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders profile image unchanged 1`] = `
<div>
  <div
    class="relative flex-shrink-0 w-10 h-10 foo bar "
  >
    <img
      alt="Profile Image"
      class="w-10 h-10 bg-contain bg-center rounded-full object-cover "
      data-nimg="1"
      decoding="async"
      height="32"
      loading="lazy"
      src="http://localhost/_next/image?url=%2Ffoo.png&w=64&q=75"
      srcset="/_next/image?url=%2Ffoo.png&w=32&q=75 1x, /_next/image?url=%2Ffoo.png&w=64&q=75 2x"
      style="color: transparent;"
      width="32"
    />
  </div>
</div>
`;

exports[`renders profile image unchanged 2`] = `
<div>
  <div
    class="relative flex-shrink-0 w-10 h-10 foo bar "
  >
    <img
      alt="Profile Image"
      class="w-10 h-10 bg-contain bg-center rounded-full object-cover "
      data-nimg="1"
      decoding="async"
      height="32"
      loading="lazy"
      src="http://localhost/_next/image?url=%2Ffoo.png&w=64&q=75"
      srcset="/_next/image?url=%2Ffoo.png&w=32&q=75 1x, /_next/image?url=%2Ffoo.png&w=64&q=75 2x"
      style="color: transparent;"
      width="32"
    />
  </div>
</div>
`;

exports[`renders profile image unchanged 3`] = `
<div>
  <div
    class="relative flex-shrink-0 w-10 h-10 foo bar "
  >
    <img
      alt="Profile Image"
      class="w-10 h-10 bg-contain bg-center rounded-full object-cover "
      data-nimg="1"
      decoding="async"
      height="32"
      loading="lazy"
      src="http://localhost/_next/image?url=%2Ffoo.png&w=64&q=75"
      srcset="/_next/image?url=%2Ffoo.png&w=32&q=75 1x, /_next/image?url=%2Ffoo.png&w=64&q=75 2x"
      style="color: transparent;"
      width="32"
    />
  </div>
</div>
`;
